### PR TITLE
Add UNKNOWN alert state in alerts page

### LINKED ISSUE #4510: Distinguish not+yet-evaluated alert from empty alert
Like we do for targets (UNKNOWN) I'd like to see alerts that have mnot yet been evaluated (after a restart or  new alerts) marked as unknown.

ne

### PR DESCRIPTION
**Pull request by sipian**
Closes #4510 
From the discussion in #4510, I didn't quite understand @brian-brazil comment about alerting rules with same name.
An alerting rule has one expr. And once it is evaluated after `evaluation_interval`, it's health will not remain unknown anymore.
Do tell if my interpretation is incorrect.
screenshot from 2018-08-29 12-41-51

Signed-off-by: Harsh Agarwal <cs15btech11019@iith.ac.in>

### CODE REVIEW

**Comment by gouthamve** [comment]
Ah, yep, even if there are alerts with same name and expr, those are shown as different alerts in UI. Could you make `UNKNOWN` state alerts to be colorless? As shown in the image in #4510?

**Comment by gouthamve** [comment]
Also should be `unknown`

File: rules/alerting.go [hunk 1]
```diff
@@ -413,6 +413,15 @@ func (r *AlertingRule) ActiveAlerts() []*Alert {
 	return res
 }
 
+// IsAlertUnknown returns whether an alert's RuleHealth is still unkown.
+func (r *AlertingRule) IsAlertUnknown() RuleHealth {
```
Comment (codesome) [inline rules/alerting.go]: Would `IsHealthUnknown` be better here? Also with such function name, I would be expecting `true` or `false`.

Comment (sipian) [reply rules/alerting.go]: @codesome sounds correct

File: rules/alerting.go [hunk 2]
```diff
@@ -413,13 +413,13 @@ func (r *AlertingRule) ActiveAlerts() []*Alert {
 	return res
 }
 
-// IsAlertUnknown returns whether an alert's RuleHealth is still unkown.
-func (r *AlertingRule) IsAlertUnknown() RuleHealth {
-	var health RuleHealth
+// IsHealthUnknown returns whether an alerts health is still unkown.
+func (r *AlertingRule) IsHealthUnknown() bool {
+	res := false
```
Comment (codesome) [inline rules/alerting.go]: You can just do `return r.Health() == HealthUnknown` :)

File: rules/alerting.go [hunk 3]
```diff
@@ -413,13 +413,13 @@ func (r *AlertingRule) ActiveAlerts() []*Alert {
 	return res
 }
 
-// IsAlertUnknown returns whether an alert's RuleHealth is still unkown.
-func (r *AlertingRule) IsAlertUnknown() RuleHealth {
-	var health RuleHealth
+// IsHealthUnknown returns whether an alerts health is still unkown.
```
Comment (codesome) [inline rules/alerting.go]: 1. *unknown.
2. Rephrasing of the comment might be needed, not quite clear, `... returns true if health ...`?

**Comment by sipian** [comment]
cc @gouthamve  @brian-brazil

**Comment by roidelapluie** [comment]
What is the status?

File: web/ui/templates/alerts.html [hunk 4]
```diff
@@ -15,8 +15,11 @@ <h1>Alerts</h1>
     {{$alertStateToRowClass := .AlertStateToRowClass}}
     {{range .AlertingRules}}
       {{$activeAlerts := .ActiveAlerts}}
-      <tr class="{{index $alertStateToRowClass .State}} alert_header">
-        <td><i class="icon-chevron-down"></i> <b>{{.Name}}</b> ({{len $activeAlerts}} active)</td>
+      {{$IsHealthUnknown := .IsHealthUnknown}}
```
Comment (codesome) [inline web/ui/templates/alerts.html]: To keep variable naming consistent and avoid confusion, I would suggest `$isHealthUnknown := .IsHealthUnknown`.

**Comment by codesome** [comment]
One minor nit above, else LGTM :+1: . Also could you fix the merge conflict? @sipian

**Comment by roidelapluie** [comment]
Any update on this?

**Comment by codesome** [comment]
Superseded by https://github.com/prometheus/prometheus/pull/6924

**Pull request closed without merging**